pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -118,6 +119,36 @@ func (ngm *nvidiaGPUManager) Capacity() v1.ResourceList {
 	}
 }
 
+func containsInt(s string) bool {
+	for _, c := range s {
+		if unicode.IsDigit(c) {
+			return true
+		}
+	}
+	return false
+}
+
+// fixed #42412, release the GPU resources
+func (ngm *nvidiaGPUManager) UpdateDevices(podUID string, exitedContainerID string) {
+	inspectJSON, err := ngm.dockerClient.InspectContainer(exitedContainerID)
+	if err != nil {
+		glog.Errorf("SyncPod Inspect JSON container:%v, error:%v", exitedContainerID, err)
```

Comment:
Well, the method can return the desired output as long as we invoke it before the container destroyed. I have followed your advice and moved it into the caller.